metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanSystemAspect.java

```diff
@@ -129,6 +164,64 @@ public EbeanSystemAspect forUpdate(
       this.ebeanAspectV2 = ebeanAspectV2;
       this.urn = UrnUtils.getUrn(ebeanAspectV2.getUrn());
       this.aspectName = ebeanAspectV2.getAspect();
+
+      // Pre-patch validation: check existing aspect size from DB
+      if (prePatchValidationConfig != null
+          && prePatchValidationConfig.getPrePatch() != null
+          && prePatchValidationConfig.getPrePatch().isEnabled()) {
+        String rawMetadata = ebeanAspectV2.getMetadata();
+        if (rawMetadata != null) {
+          long actualSize = rawMetadata.length();
+          long threshold = prePatchValidationConfig.getPrePatch().getMaxSizeBytes();
+
+          if (actualSize > threshold) {
```

Comment:
I think you'll want to pull this logic into a util class for re-use by cassandra code path.